addons/account/models/account_journal_dashboard.py

```diff
@@ -437,17 +440,53 @@ def _fill_sale_purchase_dashboard_data(self, dashboard_data):
             "account_move.invoice_date",
             "account_move.company_id",
         ]
-        query, params = sale_purchase_journals._get_open_bills_to_pay_query().select(*field_list)
-        self.env.cr.execute(query, params)
-        query_results_to_pay = group_by_journal(self.env.cr.dictfetchall())
-
-        query, params = sale_purchase_journals._get_draft_bills_query().select(*field_list)
+        payment_field_list = [
+            "account_move_line.journal_id",
+            "account_move_line.move_id",
+            "account_move_line.balance AS amount_total_company",
+        ]
+        # DRAFTS
+        query, params = sale_purchase_journals._get_draft_bills_query().select(*bills_field_list)
         self.env.cr.execute(query, params)
         query_results_drafts = group_by_journal(self.env.cr.dictfetchall())
 
-        query, params = sale_purchase_journals._get_late_bills_query().select(*field_list)
-        self.env.cr.execute(query, params)
-        late_query_results = group_by_journal(self.env.cr.dictfetchall())
+        # WAITING BILLS AND PAYMENTS
+        query_results_to_pay = {}
+        if purchase_journals:
+            query, params = purchase_journals._get_open_bills_to_pay_query().select(*bills_field_list)
+            self.env.cr.execute(query, params)
+            query_results_bills_to_pay = group_by_journal(self.env.cr.dictfetchall())
+            query, params = purchase_journals._get_open_payments_query().select(*payment_field_list)
+            self.env.cr.execute(query, params)
+            query_results_payments_to_pay = group_by_journal(self.env.cr.dictfetchall())
+        else:
+            query, params = sale_journals._get_open_bills_to_pay_query().select(*bills_field_list)
+            self.env.cr.execute(query, params)
+            query_results_bills_to_pay = group_by_journal(self.env.cr.dictfetchall())
```

Comment:
So you don't compute anything for the sales journals if you have purchase journals?